pkg/volume/validation/pv_validation.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/volume"
```

Comment:
I think I am getting vague tingling of what you are proposing, but it would be very helpful if you can show what you mean by some code snippet just for illustration.

The volume API defines interfaces which dictate what they support and what they don't.  We need to initialize concrete implementation of those interfaces to find out, if they support particular feature or not.  That is what `ProbeVolumePlugins` does - it initializes concrete implementation of volume plugins to figure out what features they support.  I don't like having to initialize the plugin to find out its capabilities and have opened a separate issue for that - https://github.com/kubernetes/kubernetes/issues/42386 but I am still thinking about how to implement that properly. May be this will help with that issue as well.